server/gfmd/db_ldap.c

```diff
@@ -3264,6 +3264,162 @@ gfarm_ldap_quota_dir_load(void *closure,
 	return (GFARM_ERR_FUNCTION_NOT_IMPLEMENTED);
 }
 
+/**********************************************************************/
+
+static char *gfarm_ldap_user_auth_make_dn(void *vkey);
+static gfarm_error_t gfarm_ldap_user_auth_set_field(void *info,
+	char *attribute, char **vals);
+
+struct gfarm_ldap_user_auth_key {
+	const char *username;
+	const char *auth_method;
+};
+
+static const struct gfarm_ldap_generic_info_ops gfarm_ldap_user_auth_ops = {
+	&gfarm_base_user_auth_ops,
```

Comment:
please replace gfarm_base_user_auth_ops by db_base_user_auth_arg_ops,
because compilation shows the following error:
```
./db_ldap.c:1559:3: error: 'gfarm_base_user_auth_ops' undeclared here (not in a function); did you mean 'gfarm_ldap_user_auth_ops'?
  &gfarm_base_user_auth_ops,
   ^~~~~~~~~~~~~~~~~~~~~~~~
   gfarm_ldap_user_auth_ops
```